department-of-veterans-affairs/caseflow
# APPEALS-61667: Modify legacy and AOJ legacy affinity filters to not distribute any appeals outside of their affinity window if not_genpop is passed

Resolves [APPEALS-61667](https://jira.devops.va.gov/browse/APPEALS-61667)

# Description
Modifies the `case_docket.rb` file to filter for only in-window affinity and tied appeals when `genpop: 'not_genpop'` is passed to the `distribute_priority_appeals` method. This will enable the push priority job to distribute only affinity or tied to cases to judges, so the `distribute_nonpriority_appeals` does not need to account for the `not_genpop` value.

Specific changes:
- Added `genpop` arg to be passed through `distribute_priority_appeals` to `distribute_appeals` and the affinity filter methods `cavc_affinity_filter`, `cavc_aod_affinity_filter`, [FUTURE UPDATE] `aoj_cavc_affinity_filter`, and `aoj_aod_affinity_filter`
- Updated various conditionals and methods in the filters to modify whether they select or reject an appeal based on the `not_genpop` value
- Added seed data
- Fixed the global sequences which rely on the current time to no longer contain leading zeroes which was causing test failures
- Add tests for calling `distribute_priority_appeals` with `genpop: 'not_genpop'`

## Acceptance Criteria
- [ ] No test failures

## Testing Plan
If you do not want to reset your local environment, run the seed file below:
1. `bundle exec rake db:seed:appeals_for_push_priority_job_distribution`

If you want to/don't care if you reset your local environment:
1. Run `make reset`

Testing the dockets:
1. Open a rails console `make c`
2. Get a push priority judge from the seed file `judge = User.find_by_css_id 'PUSHJUDGE1'`
3. Create a Legacy Docket object `ld = LegacyDocket.new`
4. Check how many appeals are available to the judge with a **dry run distribution** that does not actually distribute the appeals: `VACOLS::CaseDocket.distribute_priority_appeals(judge, "any", nil, true).to_a.count` => should return 50
5. Check how many appeals are available when getting only non-genpop: `VACOLS::CaseDocket.distribute_priority_appeals(judge, "not_genpop", nil, true).to_a.count` => should return 9
6. Create a new AOJ legacy docket object `ald = AojLegacyDocket.new`
7. Check how many appeals are available to the judge with a **dry run distribution** that does not actually distribute the appeals: `VACOLS::AojCaseDocket.distribute_priority_appeals(judge, "any", nil, true).to_a.count` => should return 72
8. Check how many appeals are available when getting only non-genpop: `VACOLS::AojCaseDocket.distribute_priority_appeals(judge, "not_genpop", nil, true).to_a.count` => should return 12
9. Create a distribution for the judge `distribution = Distribution.create!(judge: judge)`
10. Use the distribution and docket object to distribute non-genpop appeals to the judge with no limit (as they would be distributed in a push priority job): `ld.distribute_appeals(distribution, priority: true, genpop: 'not_genpop', limit: nil, style: 'push')`
11. Check the number of cases distributed `distribution.reload.distributed_cases.count` => should return 9
12. Check the number of appeals that are available when getting only non-genpop: `VACOLS::CaseDocket.distribute_priority_appeals(judge, "not_genpop", nil, true).to_a.count` => should return 0
13. Check how many appeals are available to the judge `VACOLS::CaseDocket.distribute_priority_appeals(judge, "any", nil, true).to_a.count` => should return 41 (originally 50, all non-genpop should have been distributed)
14. Use the distribution and docket object to distribute non-genpop appeals to the judge with no limit (as they would be distributed in a push priority job): `ald.distribute_appeals(distribution, priority: true, genpop: 'not_genpop', limit: nil, style: 'push')`
15. Check the number of cases distributed `distribution.reload.distributed_cases.count` => should return 21 (9 legacy, 12 aoj legacy)
16. Check the number of appeals that are available when getting only non-genpop: `VACOLS::AojCaseDocket.distribute_priority_appeals(judge, "not_genpop", nil, true).to_a.count` => should return 0
17. Check how many appeals are available to the judge `VACOLS::AojCaseDocket.distribute_priority_appeals(judge, "any", nil, true).to_a.count` => should return 60 (originally 72, all non-genpop should have been distributed)
